### ISSUE TITLE
Issues with log2

### ISSUE THREAD

**Issue by jeremiedb** [chriselrod/SLEEFPirates.jl#37]
There appears to be pending issue related #35 and #36 as experienced when using LoopVectorization. 

The following results in a error with SLEEFPirates `v0.6.41`, while it previously worked fine: 

```julia
using LoopVectorization

function lv_debug1(p)
    rho = 1.5
    metric = zero(eltype(p))
    @turbo for i in eachindex(p)
        metric += p[i]^rho
    end
    return metric
end

p = [0.1, 0.2, 0.8]
test = lv_debug1(p)
```

The above returns the following error stackstrace:
```julia
julia> test = lv_debug1(p)
ERROR: MethodError: no method matching invlog2(::Val{2})
Closest candidates are:
  invlog2(::Val{2}, ::Type{T}) where T at C:\Users\jerem\.julia\packages\SLEEFPirates\7hXWu\src\log.jl:318
  invlog2(::Val{ℯ}, ::Type{T}) where T at C:\Users\jerem\.julia\packages\SLEEFPirates\7hXWu\src\log.jl:316
  invlog2(::Val{10}, ::Type{T}) where T at C:\Users\jerem\.julia\packages\SLEEFPirates\7hXWu\src\log.jl:319
Stacktrace:
 [1] log_fast
   @ C:\Users\jerem\.julia\packages\SLEEFPirates\7hXWu\src\log.jl:283 [inlined]
 [2] log_fast
   @ C:\Users\jerem\.julia\packages\SLEEFPirates\7hXWu\src\log.jl:348 [inlined]
 [3] log2_fast
   @ C:\Users\jerem\.julia\packages\SLEEFPirates\7hXWu\src\log.jl:345 [inlined]
 [4] log2_fast
   @ C:\Users\jerem\.julia\packages\SLEEFPirates\7hXWu\src\SLEEFPirates.jl:281 [inlined]
 [5] pow_fast
   @ C:\Users\jerem\.julia\packages\VectorizationBase\0dXyA\src\special\misc.jl:80 [inlined]
 [6] macro expansion
   @ C:\Users\jerem\.julia\packages\LoopVectorization\xHfLl\src\reconstruct_loopset.jl:1107 [inlined]
 [7] _turbo_!(::Val{(false, 0, 0, 0, false, 4, 32, 15, 64, 0x0000000000000001, 1, true)}, ::Val{(:LoopVectorization, :getindex, LoopVectorization.OperationStruct(0x00000000000000000000000000000001, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, LoopVectorization.memload, 0x0001, 0x0001), :LoopVectorization, :LOOPCONSTANTINSTRUCTION, LoopVectorization.OperationStruct(0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, LoopVectorization.constant, 0x0002, 0x0000), :LoopVectorization, :pow_fast, LoopVectorization.OperationStruct(0x00000000000000000000000000000001, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000010002, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, LoopVectorization.compute, 0x0003, 0x0000), Symbol("##DROPPED#CONSTANT##"), Symbol("##DROPPED#CONSTANT##"), LoopVectorization.OperationStruct(0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000001, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, LoopVectorization.constant, 0x0004, 0x0000), :LoopVectorization, :add_fast, LoopVectorization.OperationStruct(0x00000000000000000000000000000001, 0x00000000000000000000000000000001, 0x00000000000000000000000000000000, 0x00000000000000000000000000030004, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, 0x00000000000000000000000000000000, LoopVectorization.compute, 0x0004, 0x0000))}, ::Val{(LoopVectorization.ArrayRefStruct{:y, Symbol("##vptr##_y")}(0x00000000000000000000000000000001, 0x00000000000000000000000000000001, 0x00000000000000000000000000000000, 0x00000000000000000000000000000001),)}, ::Val{(0, (5,), (2, 4), (), (), (), ())}, ::Val{(:i,)}, ::Val{Tuple{Tuple{CloseOpenIntervals.CloseOpen{Static.StaticInt{0}, Int64}}, Tuple{LayoutPointers.GroupedStridedPointers{Tuple{Ptr{Float64}}, (1,), (0,), ((1,),), ((1,),), Tuple{Static.StaticInt{8}}, Tuple{Static.StaticInt{0}}}, Float64, DataType}}}, ::Int64, ::Ptr{Float64}, ::Float64, ::LoopVectorization.StaticType{Float64})
   @ LoopVectorization C:\Users\jerem\.julia\packages\LoopVectorization\xHfLl\src\reconstruct_loopset.jl:1107
```

When using an integer as an exponent, it however works fine:

```julia
function lv_debug2(p)
    rho = 2
    metric = zero(eltype(p))
    @turbo for i in eachindex(p)
        metric += p[i]^rho
    end
    return metric
end


test = lv_debug2(p)
```

```
[[deps.LoopVectorization]]
deps = ["ArrayInterface", "ArrayInterfaceCore", "CPUSummary", "ChainRulesCore", "CloseOpenIntervals", "DocStringExtensions", "ForwardDiff", "HostCPUFeatures", "IfElse", "LayoutPointers", "LinearAlgebra", "OffsetArrays", "PolyesterWeave", "PrecompileTools", "SIMDTypes", "SLEEFPirates", "SpecialFunctions", "Static", "StaticArrayInterface", "ThreadingUtilities", "UnPack", "VectorizationBase"]
git-tree-sha1 = "c88a4afe1703d731b1c4fdf4e3c7e77e3b176ea2"
uuid = "bdcacae8-1622-11e9-2a5c-532679323890"
version = "0.12.165"

[[deps.SLEEFPirates]]
deps = ["IfElse", "Static", "VectorizationBase"]
git-tree-sha1 = "f5c896d781486f1d67c8492f0e0ead2c3517208c"
uuid = "476501e8-09a2-5ece-8869-fb82de89a1fa"
version = "0.6.41"
```

**Comment by chriselrod** [comment]
That was all really sloppy on my behalf. I didn't run the code, or wait for tests to fail and then check logs to see what he failures were.

**Comment by jeremiedb** [comment]
Problem seems fixed with v0.6.42, thanks very much for the swift fix! 
Pending erros in CI (mainly about sin / cos accuracy) are expected?

**Comment by chriselrod** [comment]
> Problem seems fixed with v0.6.42, thanks very much for the swift fix! Pending erros in CI (mainly about sin / cos accuracy) are expected?

Those accuracy tests passed once, but I've never had the time to look into them.
So, they shouldn't fail, but it is known that they do and unrelated to these issues.
